### ISSUE TITLE
feature/suggetion for add drag and drop object or text

### ISSUE THREAD

**Issue by dipenpatel235** [pedroSG94/rtmp-rtsp-stream-client-java#42]
hello..
in that can we add gif animated  object or text over live broadcasting  ?

Thanks

**Comment by pedroSG94** [comment]
Teorically yes, this is possible on stream using surface and opengl (In this library API21+).
But this need work, not easy. I'm interesing on implement a watermark that can be used for gif but no to set text. If you want set text could be added with a bitmap.

**Comment by dipenpatel235** [comment]
yes that is fine if text converted into transparent bitmap image..
overall feature intense is can able to drag any animation in live ongoing broadcasting

Hope You understand concept.
Let me know if you have any query about it.


Thanks

**Comment by pedroSG94** [comment]
Hi i did a test for it. For now it is under develop and need some fix.
To test you can donwload the project and change this lines in Default activity: 
```java
setContentView(R.layout.activity_example);
    SurfaceView surfaceView = (SurfaceView) findViewById(R.id.surfaceView);
```
To:

```java
setContentView(R.layout.activity_gl);
    OpenGlView surfaceView = (OpenGlView) findViewById(R.id.surfaceView);
```

Working with camera2 but render opengl present glitches :(

**Comment by ravi-nimavat128** [comment]
hello,

i want to pass/send GIF image in live video broadcasting on android app,but in live video we cant send both at a time.

thanks

**Comment by pedroSG94** [comment]
Gif could be added if you decode the file and send a list of frames like bitmap objects.
I can do that implementation.
To get frames maybe this can be usefull:
https://stackoverflow.com/questions/8933893/convert-each-animated-gif-frame-to-a-separate-bufferedimage

**Comment by ravi-nimavat128** [comment]
hello
javax.imageio.ImageIO is not part of the Android platform, so unfortunately
we can't use it. 
How's possible to use javax.imageio.ImageIO (packages/class) in android?
How we can send the image/GIF in live video broadcasting in android ?

Thanks.

**Comment by pedroSG94** [comment]
I found other solution to get bitmaps with GifDecoder class:
http://droid-blog.net/2011/10/15/tutorial-how-to-play-animated-gifs-in-android-%E2%80%93-part-2/

https://github.com/felipecsl/GifImageView/tree/master/library/src/main/java/com/felipecsl/gifimageview/library

**Comment by ravi-nimavat128** [comment]
hello

i set the image on the surface view but when i start the publish live video at that time just blank screen or image live(published), now i want to broadcast together(image & live video)in android.

Thanks.

**Comment by pedroSG94** [comment]
Clone the project.
If you want only watermark (overload a bitmap in the stream) you can go to default example and change on a activity this lines:
```java
    setContentView(R.layout.activity_gl);
    OpenGlView surfaceView = (OpenGlView) findViewById(R.id.surfaceView);
```

If you change that lines you will use opengl to process camera image and overload a bitmap. For now all is hardcode because this is under develop so if you want change the bitmap you can modify it in TextureManager class on encoder module.

**Comment by pedroSG94** [comment]
Gif support is available now.
If you go to OpenGlView and replace TextureManager to TextureManagerGif you can see a demo with a banana gif.
I will add a demo activity this week anyway.

**Comment by ravi-nimavat128** [comment]
okay thanks..
